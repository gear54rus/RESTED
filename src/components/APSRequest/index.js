import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, Fields, formPropTypes } from 'redux-form';
import { Panel, Form, Col, FormGroup, Checkbox, Alert } from 'react-bootstrap';
import flow from 'lodash.flow';

import BasicAuthSection from 'components/Request/BasicAuthSection';
import * as apsActions from 'store/aps/actions';
import { getError } from 'store/aps/selectors';
import { DEFAULT_APS_REQUEST } from 'constants/constants';

import Titlebar from './Titlebar';
import URLField from './URLField';
import TokenParamsField from './TokenParamsField';
import SubmitButton from './SubmitButton';

export const apsRequestForm = 'apsRequest';

function APSRequest({ tokenFetchError, handleSubmit, sendRequest }) {
  return (
    <Panel header={<Titlebar />}>
      <Form
        horizontal
        onSubmit={handleSubmit(sendRequest)}
      >
        <FormGroup>
          <Col
            xs={5}
          >
            <Field
              name="url"
              component={URLField}
            />
          </Col>

          <Col
            xs={7}
          >
            <Fields
              names={['username', 'password']}
              component={BasicAuthSection}
              small
            />
          </Col>
        </FormGroup>
        <FormGroup>
          <Col
            xs={7}
          >
            <Field
              name="type"
              component={TokenParamsField}
            />
          </Col>

          <Col
            xs={2}
          >
            <Checkbox
              disabled // to be implemented
            >
              Refresh automatically?
            </Checkbox>
          </Col>

          <Col
            xs={3}
          >
            <SubmitButton />
          </Col>
        </FormGroup>
        {tokenFetchError && (
          <Alert bsStyle="danger">
            <strong>Failed to fetch token!</strong> {tokenFetchError.message}
          </Alert>
        )}
      </Form>
    </Panel>
  );
}

APSRequest.propTypes = {
  ...formPropTypes,
  tokenFetchError: PropTypes.instanceOf(Error),
};

const formOptions = {
  form: apsRequestForm,
};

const mapStateToProps = state => ({
  initialValues: DEFAULT_APS_REQUEST,
  tokenFetchError: getError(state),
});

export default flow(
  reduxForm(formOptions),
  connect(mapStateToProps, {
    ...apsActions,
  }),
)(APSRequest);
