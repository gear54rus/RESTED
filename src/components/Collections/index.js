import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import IconButton from 'components/IconButton';
import * as Actions from 'store/collections/actions';

import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix">
      <IconButton
        onClick={() => addCollection()}
        tooltip="Add new collection"
        icon="plus"
        className="pull-right"
      />
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel>
      <Panel.Heading>
        <Panel.Title
          componentClass={Titlebar}
          addCollection={addCollection}
        />
      </Panel.Heading>
      <Panel.Body>
        <CollectionList />
      </Panel.Body>
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

