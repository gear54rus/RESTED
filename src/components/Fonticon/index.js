import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Fonticon({ icon, className }) {
  return (
    <i
      className={classNames('fas', `fa-${icon}`, className)}
      role="presentation"
    />
  );
}

Fonticon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Fonticon;
