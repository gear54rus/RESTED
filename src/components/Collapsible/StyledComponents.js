import styled from 'styled-components';
import { Button } from 'react-bootstrap';

import Fonticon from 'components/Fonticon';

export const CollapsibleHeader = styled.h4`
  margin: 0;
`;

export const AnimatedIcon = styled(Fonticon)`
  margin-left: 5px;
  transition: .25s ease-in-out;
`;

export const CollapsibleButton = styled(Button)`
  padding-left: 0;
  padding-right: 0;

  &:hover,
  &:active {
    text-decoration: none;
  }
`;
