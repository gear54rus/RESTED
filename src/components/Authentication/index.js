import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { FormSection, Field } from 'redux-form';
import { FormGroup, FormControl, Col, ControlLabel } from 'react-bootstrap';

import { textFieldShape } from 'propTypes/field';
import Collapsible from 'components/Collapsible';
import { authTypes } from 'store/auth/sagas';
import { getAuthType } from 'store/request/selectors';

import BasicFields from './BasicAuthFields';
import APSTokenFields from './APSTokenFields';
import OAuth1Fields from './OAuth1Fields';

function AuthTypeField({ input }) {
  return (
    <FormGroup controlId="auth">
      <Col
        componentClass={ControlLabel}
        xs={2}
      >
        Type
      </Col>
      <Col xs={8}>
        <FormControl
          componentClass="select"
          placeholder="Authentication type"
          {...input}
        >
          {Object.entries(authTypes).map(([type, { caption }]) => (
            <option
              key={type}
              value={type}
            >
              {caption}
            </option>
          ))}
        </FormControl>
      </Col>
    </FormGroup>
  );
}

AuthTypeField.propTypes = textFieldShape;

function Authentication({ authType }) {
  let authFields;

  switch (authType) {
    case 'basic':
      authFields = BasicFields;
      break;
    case 'apsToken':
      authFields = APSTokenFields;
      break;
    case 'oauth1':
      authFields = OAuth1Fields;
      break;
    default:
      authFields = null;
  }

  return (
    <Collapsible
      title="Authentication"
      id="auth"
    >
      <Field
        name="type"
        component={AuthTypeField}
      />
      {authFields && (
        <FormSection
          name={authType}
          component={authFields}
        />
      )}
    </Collapsible>
  );
}

Authentication.propTypes = {
  authType: PropTypes.oneOf(Object.keys(authTypes)),
};

const mapStateToProps = state => ({
  authType: getAuthType(state),
});

export default connect(mapStateToProps)(Authentication);
