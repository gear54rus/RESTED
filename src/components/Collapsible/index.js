import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Collapse, Row, Col } from 'react-bootstrap';

import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { CollapsibleButton, CollapsibleHeader, AnimatedIcon } from './StyledComponents';

export function Collapsible(props) {
  const {
    id,
    title,
    open,
    children,
    toggleCollapse,
    mountOnEnter,
    unmountOnExit,
  } = props;
  return (
    <div>
      <CollapsibleButton
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <CollapsibleHeader>
          {title}
          <AnimatedIcon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </CollapsibleHeader>
      </CollapsibleButton>
      <Collapse
        in={open}
        mountOnEnter={mountOnEnter}
        unmountOnExit={unmountOnExit}
      >
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </div>
  );
}

Collapsible.defaultProps = {
  mountOnEnter: false,
  unmountOnExit: false,
  open: false,
};

Collapsible.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  mountOnEnter: PropTypes.bool,
  unmountOnExit: PropTypes.bool,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsible);

