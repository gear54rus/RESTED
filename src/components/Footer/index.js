import React from 'react';

import { StyledFooter } from './StyledComponents';

export default function Footer() {
  return (
    <StyledFooter>
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub
        </a>
      </small>
      ·
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS/issues"
          target="_blank"
          rel="noopener noreferrer"
        >
          Report a bug
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gitter.im/RESTEDclient"
          target="_blank"
          rel="noopener noreferrer"
        >
          Chat (RESTED)
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gist.github.com/esphen/f50cc049c8ed1256a11763826dac97db"
          target="_blank"
          rel="noopener noreferrer"
        >
          Licenses (RESTED)
        </a>
      </small>
    </StyledFooter>
  );
}
