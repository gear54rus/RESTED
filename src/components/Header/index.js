import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { isDarkTheme } from 'store/options/selectors';

import { StyledHeader, BlackLink } from './StyledComponents';

export function Header({ darkMode }) {
  return (
    <StyledHeader darkMode={darkMode}>
      <h1>
        <BlackLink
          href="https://github.com/odin-public/RESTED-APS"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            alt="RESTED APS"
            className="logo"
            height="40"
            src="img/rested-aps-logo.png"
          />
          <span>RESTED APS</span>
        </BlackLink>
      </h1>
    </StyledHeader>
  );
}

Header.propTypes = {
  darkMode: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  darkMode: isDarkTheme(state),
});

export default connect(mapStateToProps)(Header);
