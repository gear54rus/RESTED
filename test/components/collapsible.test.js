import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import { Collapsible } from 'components/Collapsible';

it('should render correctly', () => {
  const tree = renderer.create(
    <Collapsible
      id="test"
      title="Test"
      toggleCollapse={() => {}}
    >
      <h3>test</h3>
    </Collapsible>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call the toggleCollapse function when activated', () => {
  const toggleCollapse = jest.fn();

  let wrapper = mount(
    <Collapsible
      open={false}
      id="test"
      title="Test"
      toggleCollapse={toggleCollapse}
    >
      <h3>test</h3>
    </Collapsible>,
  );

  expect(toggleCollapse).not.toHaveBeenCalled();

  wrapper.find('button').simulate('click');
  expect(toggleCollapse).toHaveBeenCalled();
  expect(toggleCollapse).toHaveBeenCalledWith('test', false);

  toggleCollapse.mockClear();
  wrapper = mount(
    <Collapsible
      open
      id="test2"
      title="Test"
      toggleCollapse={toggleCollapse}
    >
      <h3>test</h3>
    </Collapsible>,
  );

  expect(toggleCollapse).not.toHaveBeenCalled();

  wrapper.find('button').simulate('click');
  expect(toggleCollapse).toHaveBeenCalled();
  expect(toggleCollapse).toHaveBeenCalledWith('test2', true);
});

