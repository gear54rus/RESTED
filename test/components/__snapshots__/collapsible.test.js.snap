// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <button
    className="sc-htpNat iPErUs btn btn-link"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <h4
      className="sc-bdVaJa bcSwbK"
    >
      Test
      <i
        className="fas fa-angle-right sc-bwzfXH dSpMtd"
        role="presentation"
      />
    </h4>
  </button>
  <div
    aria-expanded={null}
    className="collapse row"
  >
    <div
      className="col-xs-12"
    >
      <h3>
        test
      </h3>
    </div>
  </div>
</div>
`;
